sysl2/sysl/ints.go

```diff
@@ -0,0 +1,220 @@
+	}
+	applications := v.m.GetApps()
+	for app := range integrations {
+		for epname, endpt := range applications[app].GetEndpoints() {
+			v.handleStatement(app, epname, endpt.GetStmt())
+		}
+	}
+	return v.drawableApps, v.apps, v.deps
+}
+
+func (v *collectApplicationDependenciesState) handleStatement(appname, epname string, stmts []*sysl.Statement) {
+	for _, stmt := range stmts {
+		switch t := stmt.GetStmt().(type) {
+		case *sysl.Statement_Call:
+			targetName := getAppName(t.Call.GetTarget())
+			if added := v.addAppDependency(appname, epname, targetName, t.Call.Endpoint, stmt); added {
+				if v.passthroughs.Contains(targetName) && t.Call.Endpoint != ".. * <- *" {
+					for epname, endpt := range v.m.GetApps()[targetName].GetEndpoints() {
+						v.handleStatement(targetName, epname, endpt.GetStmt())
+					}
+				}
+			}
+		case *sysl.Statement_Action, *sysl.Statement_Ret:
+			continue
+		case *sysl.Statement_Cond:
+			v.handleStatement(appname, epname, t.Cond.GetStmt())
+		case *sysl.Statement_Loop:
+			v.handleStatement(appname, epname, t.Loop.GetStmt())
+		case *sysl.Statement_LoopN:
+			v.handleStatement(appname, epname, t.LoopN.GetStmt())
+		case *sysl.Statement_Foreach:
+			v.handleStatement(appname, epname, t.Foreach.GetStmt())
+		case *sysl.Statement_Group:
+			v.handleStatement(appname, epname, t.Group.GetStmt())
+		case *sysl.Statement_Alt:
+			for _, choice := range t.Alt.GetChoice() {
+				v.handleStatement(appname, epname, choice.GetStmt())
+			}
+		default:
+			panic("No statement!")
+		}
+	}
+}
+
+func (v *collectApplicationDependenciesState) addAppDependency(
+	sourceApp, sourceEndpt, targetApp, targetEndpt string,
+	stmt *sysl.Statement,
+) bool {
+	if sourceEndpt == ".. * <- *" || sourceEndpt == "*" {
+		return false
+	}
+	k := fmt.Sprintf("%s:%s:%s:%s", sourceApp, sourceEndpt, targetApp, targetEndpt)
+	if _, has := v.deps[k]; has {
+		return false
+	}
+	if v.excludes.Contains(sourceApp) || v.excludes.Contains(targetApp) {
+		return false
+	}
+	if v.drawableApps.Contains(sourceApp) || v.drawableApps.Contains(targetApp) {
+		if !HasPattern(v.m.GetApps()[sourceApp].GetAttrs(), "human") {
+			v.apps.Insert(sourceApp)
+		}
+		if !HasPattern(v.m.GetApps()[targetApp].GetAttrs(), "human") {
+			v.apps.Insert(targetApp)
+		}
+	}
+	if !((v.apps.Contains(sourceApp) && v.apps.Contains(targetApp)) ||
+		(v.apps.Contains(sourceApp) && v.passthroughs.Contains(targetApp)) ||
+		v.passthroughs.Contains(sourceApp)) {
+		return false
+	}
+	v.deps[k] = AppDependency{
+		Self:      AppElement{Name: sourceApp, Endpoint: sourceEndpt},
+		Target:    AppElement{Name: targetApp, Endpoint: targetEndpt},
+		Statement: stmt,
+	}
+	return true
+}
+
+func GenerateIntegrations(
+	rootModel, title, output, project, filter, modules string,
+	exclude []string,
+	clustered, epa bool,
+) map[string]string {
+	r := make(map[string]string)
+	if filter != "" {
+		re := regexp.MustCompile(filter)
+		if !re.MatchString(output) {
+			return r
```

Comment:
filter check should be done on the variable `outputDir` on line number 141 below.

Reply:
changed